addons/project/models/project_project.py

```diff
@@ -790,22 +830,44 @@ def _get_hide_partner(self):
 
     @api.model
     def _create_analytic_account_from_values(self, values):
-        company = self.env['res.company'].browse(values.get('company_id')) if values.get('company_id') else self.env.company
+        company = self.env['res.company'].browse(values.get('company_id')) if values.get('company_id') else False
+        if company:
+            plan = company.analytic_plan_id
+            company = company.id
```

Comment:
I dont really like that change, because i end up with setting the result of the browse (which is a record) in the variable 'company_id'. Instead, i've updated the 'company_id': company to
'company_id': company.id if company else False
That way variable name are correct, and the company set on the account is clearer.